packages/flutter_driver/lib/src/driver/timeline.dart

```diff
@@ -124,11 +120,30 @@ class TimelineEvent {
 List<TimelineEvent> _parseEvents(Map<String, dynamic> json) {
   final List<dynamic> jsonEvents = json['traceEvents'] as List<dynamic>;
 
-  if (jsonEvents == null)
+  if (jsonEvents == null) {
     return null;
+  }
 
   // TODO(vegorov): use instance method version of castFrom when it is available.
-  return Iterable.castFrom<dynamic, Map<String, dynamic>>(jsonEvents)
-    .map<TimelineEvent>((Map<String, dynamic> eventJson) => TimelineEvent(eventJson))
-    .toList();
+  final List<TimelineEvent> timelineEvents =
+      Iterable.castFrom<dynamic, Map<String, dynamic>>(jsonEvents)
+          .map<TimelineEvent>(
+              (Map<String, dynamic> eventJson) => TimelineEvent(eventJson))
+          .toList();
+
+  timelineEvents.sort((TimelineEvent e1, TimelineEvent e2) {
+    final int ts1 = e1.timestampMicros;
+    final int ts2 = e2.timestampMicros;
+    if (ts1 == null && ts2 == null) {
+      return 0;
+    } else if (ts1 == null) {
+      return -1;
+    } else if (ts2 == null) {
+      return 1;
+    } else {
+      return ts1.compareTo(ts2);
```

Comment:
if nulls are not common then we are performing 3 null tests per call. Nesting could reduce that to 2.

Another idea is setting the timestampMicros to 0 (or -1) on construction? (Though it hides the fact that some events didn't have a timestamp. At least with -1, that would be an invalid value given that the field is meant to be time since beginning of tracing).

Or doing a null == 0 test on the saved values and then just falling into the compareTo method?

Or having a fake getter for nonNullTimestampMicros (or "sortingMetric")?

Reply:
What about a helper function:

```
int nullsFirstCompare(num a, num b) {
  if (a == null) {
    return b == null ? 0 : -1;
  }
  return b == null ? +1 : a.compareTo(b);
}
```

Reply:
nested them :-)